### ISSUE TITLE
NullPointerException when I try to call VirtualMachine.powerState()

### ISSUE THREAD

**Issue by StarWix** [Azure/azure-sdk-for-java#1518]
Sometimes method VirtualMachine.powerState() throws NullPointerException when I call it. 
```
java.lang.NullPointerException: null
        at com.microsoft.azure.management.compute.implementation.VirtualMachineImpl.refreshInstanceView(VirtualMachineImpl.java:220)
        at com.microsoft.azure.management.compute.implementation.VirtualMachineImpl.instanceView(VirtualMachineImpl.java:905)
        at com.microsoft.azure.management.compute.implementation.VirtualMachineImpl.powerState(VirtualMachineImpl.java:912)
```

azure sdk version: v1.0.0-beta4.1

**Comment by anuchandy** [comment]
@StarWix - you mentioned that you are getting this error intermittently. I will take a look at the code and check whether we are missing any initialization within the object in some cases. Any additional details you can provide to repro this?

**Comment by StarWix** [comment]
@anuchandy 
```
azure.virtualMachines().listByGroup(azureConfig.getResourceGroupName()).stream()
            .map(vm -> vm.powerState())
            .collect(Collectors.toList());
```
Also I noticed that NullPointerException throws when some machines are deleting and this code is executing. It happens in about 5 of 100 cases.

**Comment by martinsawicki** [comment]
I may have fixed the NPE (but i don't know how to repro the original bug, so I can't verify)

**Comment by anuchandy** [comment]
@StarWix this is fixed in PR #1603, hence closing. Let us know if you still face the issue.

**Comment by slmalone** [comment]
Hello - I am facing a similar issue where an invocation of `virtualMachine.powerState().toString()` will throw a `NullPointerException`. This seems to happen in particular when the `virtualMachine` instance in question is deleting. Even when I first check to ensure the VM is not null I will still get a `NPE`:
 
```
String state;
if (virtualMachine != null) {
    state = virtualMachine.powerState().toString();
} else {
    state = UNKNOWN;
}
```
From my `build.gradle` I'm using `implementation 'com.microsoft.azure:azure:1.35.0'`
